src/go/k8s/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -183,6 +183,37 @@ type ClusterSpec struct {
 
 	// The name of the ServiceAccount to be used by the Redpanda pods
 	ServiceAccount *string `json:"serviceAccount,omitempty"`
+
+	// NodePools runs a StatefulSet for every NodePool. All these produce pods,
+	// that join the cluster, but are steered independently (especially replica
+	// count).
+	NodePools []*NodePoolSpec `json:"nodePools,omitempty"`
```

Comment:
I'm not sure if slice of pointers to `NodePoolSpec` is necessary. It would be good to add some validation as there shouldn't be any duplicates of the same `Name` in `NodePoolSpec` slice.

Reply:
i would add validations in the implementation PR (webhook)

> I'm not sure if slice of pointers to NodePoolSpec is necessary.

what do you mean? should i use slice of non-pointer struct?

Reply:
```suggestion
	NodePools []NodePoolSpec `json:"nodePools,omitempty"`
```

I was only asking to remove pointer. But that can stay as is.